### ISSUE TITLE
`no-unnecessary-waiting` rule does not work in command chain

### ISSUE THREAD

**Issue by Lakitna** [cypress-io/eslint-plugin-cypress#21]
When writing the following the rule does not recognise the error:

```javascript
// Does not throw lint error
cy.get('foo').wait(100);
```

```javascript
// Does not throw lint error
cy.anyCommand()
  .wait(100);
```

It seems like the rule only works when `.wait()` is called as a parent command:

```javascript
// Throws lint error
cy.wait(100);
```

I think the issue is located here https://github.com/cypress-io/eslint-plugin-cypress/blob/master/lib/rules/no-unnecessary-waiting.js#L35

In fact, the implementation here is accurate when you [look at the documentation](https://docs.cypress.io/api/commands/wait.html#Rules), but not when you look at [the source code for the `.wait()` command](https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/waiting.coffee#L160).

I'll also open an issue over at `cypress-documentation` about this mismatch.
